### PR TITLE
Tokenization updated API

### PR DESCRIPTION
**Pull request by or109**
## Pull Request Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes (link to the issue here)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Chore / Documentation
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Locally tested against Fireblocks API

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added corresponding labels to the PR

### CODE REVIEW

File: src/fireblocks-sdk.ts [hunk 1]
```diff
@@ -1868,61 +1912,90 @@ export class FireblocksSDK {
...
      * Link a token to the tenant
-     * @param assetId
+     * @param type
+     * @param refId
+     *
+     * @returns TokenLink
      */
-    public async linkToken(assetId: string): Promise<TokenLink> {
-        return await this.apiClient.issuePutRequest(`/v1/tokenization/tokens/${assetId}/link`, {});
+    public async linkToken(type: SupportedContractTemplateType, refId: string): Promise<TokenLink> {
```
Comment (idanya) [inline src/fireblocks-sdk.ts:1966]: you are using just `id` in other endpoints (not `refId`)

File: src/types.ts [hunk 2]
```diff
@@ -1627,18 +1627,37 @@ export enum NFTOwnershipWalletType {
...
     name: string;
     description: string;
     longDescription: string;
     bytecode: string;
     sourcecode: string;
+    type?: ContractTemplateType;
+    implementationContractId?: string;
+    initializationPhase: ContractInitializationPhase;
     compilerOutputMetadata?: object;
+    inputFieldsMetadata?: object;
```
Comment (idanya) [inline src/types.ts:1654]: we have a DTO for this...

File: src/fireblocks-sdk.ts (hunk 1 above)
Comment (or109) [reply src/fireblocks-sdk.ts:1966]: as we talked, `id` is being generated only after the token link was created/linked and not before

File: src/types.ts (hunk 2 above)
Comment (or109) [reply src/types.ts]: `InputFieldsMetadata` interface was added

File: src/fireblocks-sdk.ts [hunk 3]
```diff
@@ -1605,15 +1611,14 @@ export class FireblocksSDK {
...
-    public async getNFTs(filter: GetNFTsFilter): Promise<Web3PagedResponse<Token>> {
-        const { pageCursor, pageSize, ids, sort, order } = filter;
+     * @param {Object} payload
+     * @param payload.pageCursor
+     * @param payload.pageSize
+     * @param payload.ids
+     * @param payload.sort
+     * @param payload.order
+     */
+    public async getNFTs({ pageCursor, pageSize, ids, sort, order }: GetNFTsFilter): Promise<Web3PagedResponse<Token>> {
```
Comment (amimagid) [inline src/fireblocks-sdk.ts:1621]: why here?

File: src/fireblocks-sdk.ts [hunk 4]
```diff
@@ -1772,20 +1777,33 @@ export class FireblocksSDK {
...
-        return await this.apiClient.issueGetRequest(`/v1/contract-registry/contracts?${queryString.stringify(requestFilter)}`);
+     * @param {Object} payload The payload for getting the current tenant's sessions
+     * @param {ContractInitializationPhase} payload.initializationPhase The contract initialization phase
+     * @param {ContractTemplateType} payload.type The type of the contract templates you wish to retrieve. May contain more than one type
+     * @param {number} payload.pageSize The amount of results to return on the next page
+     * @param {string} payload.pageCursor The cursor for the next page
+     *
+     * @returns {LeanContractTemplateDto[]} A paginated array of contract templates
+     */
+    public async getTemplateContracts({
```
Comment (amimagid) [inline src/fireblocks-sdk.ts:1788]: let's rename to getContractTemplates

File: src/fireblocks-sdk.ts [hunk 5]
```diff
@@ -1772,20 +1777,33 @@ export class FireblocksSDK {
...
+        })}`);
     }
 
     /**
      * Upload a new contract. This contract would be private and only your tenant can see it
      * @param payload
+     *
+     * @returns {ContractTemplateDto}
      */
     public async uploadTemplateContract(payload: ContractUploadRequest): Promise<ContractTemplateDto> {
```
Comment (amimagid) [inline src/fireblocks-sdk.ts:1808]: uploadContractTemplate

File: src/fireblocks-sdk.ts [hunk 6]
```diff
@@ -1794,6 +1812,8 @@ export class FireblocksSDK {
     /**
      * Get contract template by id
      * @param contractId
+     *
+     * @returns {ContractTemplateDto}
      */
     public async getTemplateContract(contractId: string): Promise<ContractTemplateDto> {
```
Comment (amimagid) [inline src/fireblocks-sdk.ts:1818]: getContractTemplate

File: src/fireblocks-sdk.ts [hunk 7]
```diff
@@ -1811,6 +1831,8 @@ export class FireblocksSDK {
      * Get contract template constructor by contract id
      * @param contractId
      * @param withDocs
+     *
+     * @returns {ContractTemplateDto}
      */
     public async getTemplateContractConstructor(contractId: string, withDocs: boolean = false): Promise<ContractTemplateDto> {
```
Comment (amimagid) [inline src/fireblocks-sdk.ts:1837]: getContractTemplateConstructor

File: src/fireblocks-sdk.ts [hunk 8]
```diff
@@ -1868,61 +1912,90 @@ export class FireblocksSDK {
...
-     * Issue a new token and link it to the tenant
-     * @param payload
+     * Creates a new asset and links it as a token.
+     *
+     * @param {IssueTokenRequest} payload - The payload containing information for token issuance.
+     *
+     * @returns {CreateTokenResponseDto} Response with created token ID
      */
-    public async issueNewToken(payload: IssueTokenRequest): Promise<PendingTokenLinkDto> {
+    public async issueNewToken(payload: IssueTokenRequest): Promise<CreateTokenResponseDto> {
```
Comment (amimagid) [inline src/fireblocks-sdk.ts:1929]: Wait... why not return `TokenLinkDto`? with Pending status? how will ripple and stellar work? you need to do another request?

File: src/fireblocks-sdk.ts (hunk 3 above)
Comment (or109) [reply src/fireblocks-sdk.ts:1621]: keeping the same convention of using pagination filters

File: src/fireblocks-sdk.ts (hunk 8 above)
Comment (or109) [reply src/fireblocks-sdk.ts:1929]: It's possible

Let's decide it tm and have the right change for it 🙏

File: src/fireblocks-sdk.ts (hunk 3 above)
Comment (amimagid) [reply src/fireblocks-sdk.ts:1621]: it's just distructing on the signature level and not within the function. what's the convention that is kept?

Comment (or109) [reply src/fireblocks-sdk.ts:1622]: destruction in the signature 🙏🏻

**Review by amimagid — APPROVED** [review]

**Pull request merged**